portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -80,16 +90,20 @@ public void createEntitys(){
     datastore.put(commentEntity);
   }
 
-  public void populateList(PreparedQuery results){
+  public void displayMaxComments(PreparedQuery results, int maxComment){
     this.comments = new ArrayList<>();
-    for (Entity commentEntity : results.asIterable()) {
+    for (Entity commentEntity : results.asIterable()){
+      if(maxComment == 0){
+          break;
+      }
       Comment comment = new Comment(
         commentEntity.getKey().getId(), 
         (String) commentEntity.getProperty(USERNAME_PARAMETER), 
```

Comment:
I think you should be able to use a .toString() method, as it worked for me.

Reply:
+1

Casting is dangerous and leads to undefined behavior. The `toString` method is well defined